Automattic/wp-calypso
# Checkout: Add thank you page for the domain only site flow

Implements #11452. ~~Requires D4612-code.~~

This PR:

- Creates `ThankYouCard` from `PlanThankYouCard`.
- Refactors `PlanThankYouCard` to use `ThankYouCard`.
- Uses `ThankYouCard` to display a domain-only thank you page.

### Testing
*`PlanThankYouCard`*
- Make sure https://wpcalypso.wordpress.com/devdocs/blocks/plan-thank-you-card and http://calypso.localhost:3000/devdocs/blocks/plan-thank-you-card looks the same after refactoring. See also image below for comparison.
![screen shot 2017-03-02 at 17 39 49](https://cloud.githubusercontent.com/assets/699132/23517330/4f6d7746-ff70-11e6-8a62-dedddad8d8f1.png)

- Visit `/start` while logged out (`PlanThankYouCard` is only displayed for new users).
- Select a non-free plan on the plans page.
- Proceed through signup and create a new user.
- Assert that you land in checkout with the plan you selected in the second step.
- Purchase the plan.
- Assert that you land on a thank you page with `PlanThankYouCard` displayed for the plan you purchased like in the image above.

*New thank you page*
- Visit `/start/domain-first?new=DOMAIN` where `DOMAIN` is a domain like `thisisaniquedomain123.live`.
- Select `Just purchase a domain`.
- Purchase the domain in checkout.
- Assert that you land on a thank you page with `ThankYouCard` displayed for the domain you purchased like:
![screen shot 2017-03-06 at 12 21 00](https://cloud.githubusercontent.com/assets/699132/23608603/72193b5c-026a-11e7-85f7-40907b4be8de.png)


### Review
